docs/design_docs/index_design.md

```diff
@@ -0,0 +1,191 @@
+information corresponding to each IndexNode node, and update the priority queue in the NodeManager. Whenever the task
+of building an index needs to be allocated, the IndexNode with the lowest load will be selected according to the 
+priority queue to execute the task.
+
+### 8.3.3 MetaTable
+
+In order to maintain the status information of the index, we introduced MetaTable to record the status information
+of the index. In order to ensure that the MetaTable information is not lost after IndexCoord is powered off and
+restarted, we write the MetaTable information into ETCD. When the IndexCoord service starts, it will first load the
+existing Meta information from ETCD, and then monitor the changes of Meta through watchNodeLoop. In order to distinguish
+whether the modification of Meta was initiated by IndexCoord or IndexNode, revision was introduced in Meta.
+When watchMetaLoop detects that the Meta in ETCD is updated, compare the revision in Meta with the Event.Kv.Version
+of the ETCD event. If the revision is equal to Event.Kv.Version, it means that the update was initiated by IndexCoord.
+If the revision is less than Event. .Kv.Version means that this Meta update was initiated by IndexNode, and IndexCoord
+needs to update Meta. There will be no situation where revision is greater than Event.Kv.Version.
+
+In order to prevent IndexNode from appearing in a suspended animation state, Version is introduced. When IndexCoord 
+finds that IndexNode is offline, it assigns the unfinished tasks that IndexNode is responsible for to other IndexNodes,
+and adds 1 to Version. After the task is completed, it is found that the version corresponding to the task is already
+larger than the version corresponding to the task it is executing, and the Meta is not updated.
+
+### 8.3.4 watchNodeLoop
+
+`watchNodeLoop` is used to monitor IndexNode going online and offline. When IndexNode goes online and offline, 
+IndexCoord adds or deletes the corresponding IndexNode information in NodeManager.
+
+### 8.3.5 watchMetaLoop
+
+`watchMetaLoop` is used to monitor whether the Meta in ETCD has been changed. When the Meta in the ETCD is monitored,
+the result of the Meta update is obtained from the ETCD, and the `Event.Kv.Version` of the update event is compared
+with the `revision` in the MetaTable. If the `Event.Kv.Version` is greater than the `revision` in the MetaTable, 
+Explain that this update is initiated by IndexNode, and then update the MetaTable in IndexCoord. Since this update
+is initiated by IndexNode, it indicates that this IndexNode has completed this task, so update the load of this 
+IndexNode in NodeManager, and the task amount is reduced by one.
+
+### 8.3.6 assignTaskLoop
+
+`assignTaskLoop` is used to assign index construction tasks. There is a timer here to traverse the MetaTable regularly
+to filter out the tasks that need to be allocated, including unallocated tasks and tasks that have been suspended by
```

Comment:
including unallocated tasks and tasks that have been failed due to indexNode crash